Aplib.Core/Desire/Goal.cs

```diff
@@ -0,0 +1,79 @@
+namespace Aplib.Core.Desire
+{
+    public class Goal
+    {
+        /// <summary>
+        /// The goal is considered to be completed, when the distance of the
+        /// <see cref="CurrentHeuristics"/> is below this value.
+        /// </summary>
+        private const float Epsilon = 0.005f;
+
+        /// <summary>
+        /// The <see cref="Heuristics"/> of the current state of the game.
+        /// </summary>
+        public Heuristics CurrentHeuristics { get; private set; }
+
+        /// <summary>
+        /// The <see cref="Tactic"/> used to achieve this <see cref="Goal"/>, which is executed during every iteration
+        /// of the BDI cycle.
+        /// </summary>
+        /// <seealso cref="Iterate()"/>
+        private readonly Tactic _tactic;
+        /// <summary>
+        /// The <see cref="IGoalPredicate"/> used to test whether this <see cref="Goal"/> has been completed.
+        /// </summary>
+        /// <seealso cref="IsCompleted()"/>
+        private readonly IGoalPredicate _goalPredicate;
+
+        // MetaData useful for debugging
+        /// <summary>
+        /// The name used to display the current goal during debugging, logging, or general overviews.
+        /// </summary>
+        public readonly string Name;
+        /// <summary>
+        /// The description used to describe the current goal during debugging, logging, or general overviews.
+        /// </summary>
+        public readonly string Description;
```

Comment:
Make these fields auto-properties. It is uncommon to have `public readonly` fields, as auto-properties are a newer feature and thus should be used.

They also enable quick alteration of the visibility of getters and setters.
Also: an enter was forgotten between the fields.

```suggestion
        // MetaData useful for debugging
        /// <summary>
        /// The name used to display the current goal during debugging, logging, or general overviews.
        /// </summary>
        public string Name { get; }

        /// <summary>
        /// The description used to describe the current goal during debugging, logging, or general overviews.
        /// </summary>
        public string Description { get; }
```